modules/routes/Forums/forum_route.py

```diff
@@ -0,0 +1,147 @@
+def create_course_forum(course_code):
+    # Check if the user is a student
+    session = fetch_session()
+    db_cursor = db.cursor(dictionary=True)
+    if session["account_type"] == AccountType.Student.name:
+        return jsonify({"message": "Students are not allowed to create forums"}), 403
+
+    # Check if the user is a lecturer who teaches this course
+    if session["account_type"] == AccountType.Lecturer.name:
+        # Check if the lecturer teaches the specified course
+        lecture_details = db_cursor.execute(
+            "SELECT * FROM Course WHERE lecturer_id = %s", (session["sub"],)
+        ).fetchone()
+        if course_code != lecture_details["course_code"]:
+            return jsonify({"message": "You can only view your own courses!"}), 403
+
+
+    # Proceed with forum creation logic
+    body = ForumSchema().load(request.get_json(force=True))
+    body["course_code"] = course_code  
+
+    db_cursor = db.cursor()
+    try:
+        db_cursor.execute(
+            "INSERT INTO DiscussionForum (topic, post_time, creator, course_code) VALUES (%s, %s, %s, %s)",
+            (body["topic"], date.today(), fetch_session()["sub"], body["course_code"]),
+        )
+        db.commit()
+        return jsonify({"message": "Forum created successfully"}), 201
+    except Exception as e:
+        db.rollback()
+        return jsonify({"message": f"Failed to create forum: {str(e)}"}), 500
+
+
+@app.route("/threads/forum/<int:forum_id>", methods=["GET", "POST"])
+@protected_route(roles=[AccountType.Admin, AccountType.Lecturer])
+def handle_forum_threads(forum_id, course_code):
+    # Check if the user is a student
+    session = fetch_session()
+    db_cursor = db.cursor(dictionary=True)
+    if session["account_type"] == AccountType.Student.name:
+        return jsonify({"message": "Students are not allowed to create forums"}), 403
+
+    # Check if the user is a lecturer who teaches this course
+    if session["account_type"] == AccountType.Lecturer.name:
+        # Check if the lecturer teaches the specified course
+        lecture_details = db_cursor.execute(
+            "SELECT * FROM Course WHERE lecturer_id = %s", (session["sub"],)
+        ).fetchone()
+        if course_code != lecture_details["course_code"]:
+            return jsonify({"message": "You can only view your own courses!"}), 403
+
+    # Proceed with retrieving discussion threads for the forum
+    db_cursor.execute("SELECT * FROM DiscussionThread WHERE forum_id = %s", (forum_id,))
+    threads = db_cursor.fetchall()
+    if not threads:
+        return jsonify({"message": "No discussion threads found for this forum"}), 404
+    # Convert the threads data into JSON format and return
+    return jsonify(threads), 200
+
+@app.route("/threads/forum/<int:forum_id>", methods=["POST"])
+@protected_route(roles=[AccountType.Admin, AccountType.Lecturer, AccountType.Student])
+def add_thread_to_forum(forum_id, course_code):
```

Comment:
You didn't even use a course code?

```suggestion
def add_thread_to_forum(forum_id: int):
```